src/TTrigger.cpp

```diff
@@ -275,6 +284,7 @@ bool TTrigger::match_perl(char* subject, const QString& toMatch, int regexNumber
     assert(mRegexMap.contains(regexNumber));
```

Comment:
:thought_balloon: It seems we might have been predicting the sort of duplication we need to do (IMHO) if `subject` and `toMatch` are, in fact, the *same* content, but in different encodings (UTF-8 vs UTF16{BE}) - could it be that all of these `match_xxx(...)` functions could be trimmed down to only take ONE of them - and we regenerate the other one as needed internally?

Alternatively, it would be clearer to rename one of them to be more like the other - perhaps `subject16` for the second one, to emphasis that they ARE the same data but presented in a different way? Mind you - not every coder will recognise that `subject` ***is*** what is being fed into the PCRE(8/16) libraries to be matched against the compiled regex pattern so `toMatch8` and `toMatch16` is more comprehensible - note that using a numeric suffix on BOTH will make for a fairly simple global search-and-replace as you'd convert all existing `toMatch`s to `toMatch16`s *then* all `subject`s to `toMatch8`s ...!

Reply:
That seems reasonable, but I'm not sure if it wasn't better to do it in separate commit.